invenio_rdm_migrator/streams/communities/table_generator.py

```diff
@@ -17,17 +18,41 @@ def _generate_members_uuids(data):
     return data["community_members"]
 
 
+def _generate_featured_community_id(data):
+    if not data.get("featured_community"):
+        return None
```

Comment:
I am not a big fan of this approach, feel free to suggest alternatives.

The idea here is that some communities are not featured, therefore the PK for `featured_community` should not be generated at all.

Reply:
I have a similar case on files migration and I have moved the id generation on the transformation layer i.e on the entry class. In my case, it was a uuid that needed to be generated so I didn't need a cache but might be that you can just use the `_pid_pk` function there... Probably still not the best solution but at least on the transformation layer you anticipate to make these checks or probably you know already if the community is featured or not before you react the generation of this pk...We can discuss this IRL if it's easier :)